src/models.py

```diff
@@ -1,81 +1,64 @@
-from functools import partial
-
 import torch
-import torch.nn as nn
 from torch.autograd import Variable
 from torch import optim
-import pandas as pd
-from sklearn.model_selection import train_test_split
-from sklearn.externals import joblib
-from sklearn.ensemble import RandomForestRegressor
-from toolkit import Model
 
-from .callbacks import NeptuneMonitorSegmentation, ValidationMonitorSegmentation
-from .steppy.pytorch.architectures.unet import UNet
 from .steppy.pytorch.callbacks import CallbackList, TrainingMonitor, ModelCheckpoint, \
-    ExperimentTiming, ExponentialLRScheduler, EarlyStopping
+    ExperimentTiming, ExponentialLRScheduler, EarlyStopping, NeptuneMonitor, ValidationMonitor
 from .steppy.pytorch.models import Model
-from .steppy.pytorch.validation import multiclass_segmentation_loss, DiceLoss
-from .steppy.sklearn.models import LightGBM, make_transformer, SklearnRegressor
-from .utils import softmax
-from .unet_models import AlbuNet, UNet11, UNetVGG16, UNetResNet
-
 
+from .retinanet import RetinaNet, RetinaLoss
 
 
-class BaseRetina(Model):
+class Retina(Model):
     def __init__(self, architecture_config, training_config, callbacks_config):
         """
         """
         super().__init__(architecture_config, training_config, callbacks_config)
+        self.encoder_depth = self.architecture_config['model_params']['encoder_depth']
+        self.num_classes = self.architecture_config['model_params']['num_classes']
+        self.pretrained = self.architecture_config['model_params']['pretrained']
         self.set_model()
-        self.weight_regularization = weight_regularization_unet
+        self.weight_regularization = weight_regularization
         self.optimizer = optim.Adam(self.weight_regularization(self.model, **architecture_config['regularizer_params']),
                                     **architecture_config['optimizer_params'])
-        self.loss_function = None
-        self.callbacks = callbacks_unet(self.callbacks_config)
-
-    def fit(self, datagen, validation_datagen=None, meta_valid=None):
-        self._initialize_model_weights()
+        self.loss_function = [('FocalLoss', RetinaLoss(num_classes=self.num_classes), 1.0)]
+        self.callbacks = callbacks(self.callbacks_config)
 
-        self.model = nn.DataParallel(self.model)
-
-        if torch.cuda.is_available():
-            self.model = self.model.cuda()
-
-        self.callbacks.set_params(self, validation_datagen=validation_datagen, meta_valid=meta_valid)
-        self.callbacks.on_train_begin()
+    def transform(self, datagen, *args, **kwargs):
+        self.model.eval()
```

Comment:
@taraspiotr @jakubczakon I suggest to get rid of transforming in train mode. Just add `train_mode` as a parameter in `__init__` and write:
```
def transform(self, datagen, *args, **kwargs):
   if self.train_mode:
      return self
```
or something similiar. It's just time-saving, because transform may take a loooot of time with this dataset. Do you like it?

Reply:
@apyskir I agree and it's a nice way to do it